MQ2AutoSize.cpp

```diff
@@ -331,13 +497,24 @@ PLUGIN_API void OnEndZone()
 
 PLUGIN_API void OnPulse()
 {
-	if (GetGameState() != GAMESTATE_INGAME || !AS_Config.OptByRange) return;
+	if (GetGameState() != GAMESTATE_INGAME)
+	{
+		return;
+	}
+
 	if (uiSkipPulse < SKIP_PULSES)
 	{
 		uiSkipPulse++;
 		return;
 	}
 
+	// check if communication plugins are still running and adjust UI as needed
+	if (GetTickCount64() > commsCheck)
+	{
+		commsCheck = std::int64_t(commsCheck) + (static_cast<long long>(commsRefreshRateSeconds) * 1000);
+		ChooseInstructionPlugin();
```

Comment:
Oh geeze, i see what you're doing.

Don't do this. Just Implement OnLoadPlugin/OnUnloadPlugin plugin callbacks

```
PLUGIN_API void OnLoadPlugin(const char* pluginName)
{
    if (ci_equals(pluginName, "MQ2EQBC"))
    {
        // ... we just loaded MQ2EQBC
    }
}
```

etc

Reply:
To work correctly I need the change I shared with you on Discord to be implemented in MQPluginHandler.cpp. I'll resolve this once that change is available.